substreams/ethereum-maverick/src/modules.rs

```diff
@@ -0,0 +1,456 @@
+                        )| {
+                            let tokens = tokens_store
+                                .get_last(format!("pool:{}", hex::encode(log.address())))
+                                .unwrap()
+                                .split(":")
+                                .map(|token| token.to_owned()) // Clone the tokens
+                                .collect::<Vec<_>>();
+                            let neg_delta_a: BigInt = delta_a * -1;
+                            let neg_delta_b: BigInt = delta_b * -1;
+                            vec![
+                                BalanceDelta {
+                                    ord: log.log.ordinal,
+                                    tx: Some(tx_from_log(&log)),
+                                    token: hex::decode(tokens[0].clone()).unwrap(),
+                                    delta: neg_delta_a.to_signed_bytes_be(),
+                                    component_id: log.address().into(),
+                                },
+                                BalanceDelta {
+                                    ord: log.log.ordinal,
+                                    tx: Some(tx_from_log(&log)),
+                                    token: hex::decode(tokens[1].clone()).unwrap(),
+                                    delta: neg_delta_b.to_signed_bytes_be(),
+                                    component_id: log.address().into(),
+                                },
+                            ]
+                        },
+                    )
+                    .collect::<Vec<_>>()
+            } else {
+                vec![]
+            }
+        })
+        .collect::<Vec<_>>();
+
+    Ok(BlockBalanceDeltas { balance_deltas: deltas })
+}
+
+/// It's significant to include both the `pool_id` and the `token_id` for each balance delta as the
+///  store key to ensure that there's a unique balance being tallied for each.
+#[substreams::handlers::store]
+pub fn store_balances(deltas: BlockBalanceDeltas, store: StoreAddBigInt) {
+    store_balance_changes(deltas, store)
+}
+
+/// This is the main map that handles most of the indexing of this substream.
+/// Every contract change is grouped by transaction index via the `transaction_contract_changes`
+///  map. Each block of code will extend the `TransactionContractChanges` struct with the
+///  cooresponding changes (balance, component, contract), inserting a new one if it doesn't exist.
+///  At the very end, the map can easily be sorted by index to ensure the final
+/// `BlockContractChanges`  is ordered by transactions properly.
+#[substreams::handlers::map]
+pub fn map_protocol_changes(
+    block: eth::v2::Block,
+    grouped_components: BlockTransactionProtocolComponents,
+    deltas: BlockBalanceDeltas,
+    components_store: StoreGetInt64,
+    balance_store: StoreDeltas, // Note, this map module is using the `deltas` mode for the store.
+) -> Result<BlockContractChanges> {
```

Comment:
```suggestion
) -> Result<BlockChanges> {
```

`BlockContractChanges` is deprecated and will be removed soon